AutoRest/Generators/Java/Java/JavaCodeNamer.cs

```diff
@@ -348,7 +348,7 @@ public static String ImportedFrom(PrimaryType primaryType)
             else if (primaryType == PrimaryType.TimeSpan ||
                 primaryType.Name == "Period")
             {
-                return "java.time.Period";
+                return "org.joda.time.Period";
```

Comment:
An ISO8601 duration is described here: https://en.wikipedia.org/wiki/ISO_8601#Durations

Looking at: http://stackoverflow.com/questions/2653567/joda-time-whats-the-difference-between-period-interval-and-duration

> An interval in Joda-Time represents an interval of time from one millisecond instant to another instant. Both instants are fully specified instants in the datetime continuum, complete with time zone. Specific times are defined e.g. this might be the interval between 20:00:00GMT yesterday and 09:00:00GMT this morning.

So it's pretty clear that we don't want an interval, as a "Duration" does not have a start and an end, so interval doesn't apply.

> A duration in Joda-Time represents a duration of time measured in milliseconds. The duration is often obtained from an interval. i.e. we can subtract start from end of an interval to derive a duration

We could choose a joda Duration... 

> A period in Joda-Time represents a period of time defined in terms of fields, for example, 3 years 5 months 2 days and 7 hours. This differs from a duration in that it is inexact in terms of milliseconds. A period can only be resolved to an exact number of milliseconds by specifying the instant (including chronology and time zone) it is relative to. e.g. consider the period of 1 year, if we add this to January 1st we will always arrive at the next January 1st but the duration will depend on whether the intervening year is a leap year or not. Similarly if we add 1 month to the 1st of a month then we will arrive at the 1st of the next month but the duration (in milliseconds) will vary based on the month in question

Period matches most closely with the description of ISO8601 I think, (ISO8601 is Years/Months/Weeks/Days/Hours/Minutes/Seconds as well).  In Joda, a Duration only has these constructors:

> Duration(long duration)
> Duration(long startInstant, long endInstant) 

But a Period has this:

> Period(int years, int months, int weeks, int days, int hours, int minutes, int seconds, int millis)

That lead me to choose Period over Duration.  If you think we should use Duration instead I can change to that.